configure.ac

```diff
@@ -870,7 +870,7 @@ AM_CONDITIONAL(OPENLDAP,test "$with_openldap" = "yes")
 
 # check for --with-libldap-r
 AC_MSG_CHECKING(for --with-libldap-r)
-AC_ARG_WITH(libldap-r, AS_HELP_STRING([--with-libldap-r],[Use libldap_r.so shared library (default: if OpenLDAP version is less than 2.6, then libldap_r.so will be used, else - libldap.so)]),
+AC_ARG_WITH(libldap-r, AS_HELP_STRING([--with-libldap-r],[Use libldap_r.so shared library (default: if OpenLDAP version is less than 2.5, then libldap_r.so will be used, else - libldap.so)]),
```

Comment:
This relies on how the distro is packaging 2.5 though because libldap.so is from 2.6 not 2.5, and that would require the lib link to be present. So I think this needs to handle that case?

Reply:
This change is about lldap_r and lldap only.

In the Upstream, libldap_r has been merged with libldap starting from openldap-2.5:
https://lists.openldap.org/hyperkitty/list/openldap-announce@openldap.org/thread/BH3VDPG6IYYF5L5U6LZGHHKMJY5HFA3L/

So lldap_r should be used before 2.5. (And lldap will be used for 2.5 and newer)

But you are correct that libldap.so and  libldap_r.so wording is wrong (as it refers to 2.6 change). I'll change the comment.

Reply:
I thought that lldap.so in 2.5 still had a version, such as lldap-2_5.so?

Reply:
> I thought that lldap.so in 2.5 still had a version, such as lldap-2_5.so?

I mean `-lldap` and `-lldap_r` parameters that we supply to GCC.
This commit https://github.com/389ds/389-ds-base/commit/a3375a08a506c3b72a8a23df5238848017691350 doesn't touch anything *.so related so the OpenLDAP 2.6 change is not touched.
And if I understood you correctly...